MetaCell/cloud-harness
# #451: model and types

Closes #451

Implemented solution: model stubs have been added based on OpenApi specification, and these models have been used to refactor the relevant apis to include typing hints.
The solution is backwards compatible as the new models can be used with the dictionary api as well.

How to test this PR: 
Most of the new functionality is test covered to be backwards compatible. Functionality based on the cloudharness Python library should work untouched if developed following the main develoment templates. The main functionality that may be affected is data serialization/deserialization.

Even if everything has been done to make this backwards compatible, it's a quite of a radical change so additional attention is needed on deployments.
The main change is the new dependency introduced on the cloudharness_models library: the installation process of cloudharness has been improved to accommodate the new library (and other future library additions).
- Codefresh deployment files need to be regenerated as the cloudharness installation is changed
- The new dependency may break Docker images depending on the Cloudharness Python library


### Sanity checks:

### Breaking changes (select one):

### Possible deployment updates issues (select one):

### Test coverage (select one):

### Documentation (select one):

### Nice to have (if relevant):
